python/ray/autoscaler/_private/aliyun/node_provider.py

```diff
@@ -0,0 +1,288 @@
+import random
+import threading
+from collections import defaultdict
+import logging
+import time
+from typing import Any, Dict, List, Optional
+
+from ray.autoscaler.node_provider import NodeProvider
+from ray.autoscaler.tags import TAG_RAY_CLUSTER_NAME, TAG_RAY_NODE_NAME, \
+    TAG_RAY_LAUNCH_CONFIG, TAG_RAY_NODE_KIND, \
+    TAG_RAY_USER_NODE_TYPE, TAG_RAY_NODE_STATUS
+from ray.autoscaler._private.constants import BOTO_MAX_RETRIES
+from ray.autoscaler._private.log_timer import LogTimer
+
+from ray.autoscaler._private.cli_logger import cli_logger
+
+from ray.autoscaler._private.aliyun.utils import AcsClient
+from ray.autoscaler._private.aliyun.config import PENDING, STOPPED, \
+    STOPPING, RUNNING, bootstrap_aliyun
+
+logger = logging.getLogger(__name__)
+
+TAG_BATCH_DELAY = 1
+STOPPING_NODE_DELAY = 1
+
+
+class AliyunNodeProvider(NodeProvider):
+    def __init__(self, provider_config, cluster_name):
+        NodeProvider.__init__(self, provider_config, cluster_name)
+        self.cache_stopped_nodes = provider_config.get("cache_stopped_nodes",
```

Comment:
Is this used at all?

Reply:
Sorry, I am busy in a training recently, my colleague chenk008 will help me fix it.

Reply:
I have fix this bug,  `cache_stopped_nodes` indicates whether delete node when node terminated